### PR TITLE
#261: refactor sync_location test

### PR DESCRIPTION
**Pull request by vladarefiev**
PR for #261:
- moved `celery.py` from sync module to root project, because it will be used for all project, not only sync.
- refactor `test_sync_location` to use `sync_locations`  task, which was already implemented in https://github.com/timelesslounge/timelessis/pull/420

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#470](https://github.com/timelesslounge/timelessis/issues/470) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#470](https://github.com/timelesslounge/timelessis/issues/470) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/9ae69d5d-f959-4388-b2c2-d0cbaef42436); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/470#issuecomment-469446655))

@paulodamaso This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#470](https://github.com/timelesslounge/timelessis/issues/470) is [in scope](http://www.zerocracy.com/policy.html#14) for 3hr
  * The role is `REV`
  * The job is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) for [3hr](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @vladarefiev[/z](https://www.0crat.com/u/vladarefiev): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23470%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: timeless/__init__.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from timeless.cache import CACHE
 from timeless.mail import MAIL
 from timeless.db import DB
-from timeless.sync.celery import make_celery
```
Comment (amihaiemil) [inline timeless/__init__.py]: @vladarefiev Why did you move this file? If it's only used for syncing, maybe it makes more sense to stay under ``timeless.sync``. or?

File: tests/integration/sync/it_sync_location_test.py [hunk 2]
```diff
@@ -49,31 +33,27 @@ def test_sync_location(db_session):
...
-        location=location,
-        poster_sync=Poster(
-            url="http://localhost:{port}".format(port=port)
-        ),
-        db_session=db_session
-    ).sync()
-    row = Location.query.filter_by(id=location.id).one()
+
+    auth_mock.return_value = 'token'
+    locations_mock.return_value = [{
```
Comment (amihaiemil) [inline tests/integration/sync/it_sync_location_test.py]: @vladarefiev don't we have a location factory for mocks?

File: tests/integration/sync/it_sync_location_test.py [hunk 3]
```diff
@@ -49,31 +33,27 @@ def test_sync_location(db_session):
...
+            "country": "United States",
+            "region": "East Coast",
+            "city": "Edge City",
+            "address": "Blvd. Kukulcan Km 9.5 #30, Plaza Forum",
+            "longitude": 21.1326063,
+            "latitude": -86.7473191,
+            "type": "L",
+            "status": "open",
+            "comment": "Nightclub from a famous movie"
+        }]
```
Comment (amihaiemil) [inline tests/integration/sync/it_sync_location_test.py]: @vladarefiev if this manual mock remains, it needs -4 spaces indentation :)

File: tests/integration/sync/it_sync_location_test.py [hunk 4]
```diff
@@ -20,27 +21,10 @@
...
-                "address": "Blvd. Kukulcan Km 9.5 #30, Plaza Forum",
-                "longitude": 21.1326063,
-                "latitude": -86.7473191,
-                "type": "L",
-                "status": "open",
-                "comment": "Nightclub from a famous movie"
-            }
-        ]
-    )
+@pytest.mark.skip
```
Comment (amihaiemil) [inline tests/integration/sync/it_sync_location_test.py]: @vladarefiev why is this test skipped?

File: tests/integration/sync/it_sync_location_test.py [hunk 5]
```diff
@@ -20,27 +21,10 @@
...
+@unittest.mock.patch.object(Authenticated, 'auth')
+@unittest.mock.patch.object(Poster, 'locations')
+def test_sync_location(locations_mock, auth_mock, db_session):
```
Comment (amihaiemil) [inline tests/integration/sync/it_sync_location_test.py]: @vladarefiev I don't see any mocking of the data actually comming from Poster. Seems like this test is incomplete?

File: timeless/__init__.py (hunk 1 above)
Comment (vladarefiev) [reply timeless/__init__.py]: @amihaiemil Actually it is not. `celery` is framework for async jobs.  particularly `celery.py` should consist of settings for jobs/queues and may be schedular for them. So, seems more logic to put in root. But you are right, every modular may contain file with jobs/tasks, usually it called `tasks.py`, we have already had this in poster module.

File: tests/integration/sync/it_sync_location_test.py (hunk 2 above)
Comment (vladarefiev) [reply tests/integration/sync/it_sync_location_test.py]: @amihaiemil I wanted to do it, but saw that there is puzzle for it.

File: tests/integration/sync/it_sync_location_test.py (hunk 3 above)
Comment (vladarefiev) [reply tests/integration/sync/it_sync_location_test.py]: @amihaiemil good catch ;)

File: tests/integration/sync/it_sync_location_test.py (hunk 4 above)
Comment (vladarefiev) [reply tests/integration/sync/it_sync_location_test.py]: @amihaiemil this test is skipped because these puzzles are not resolved yet - https://github.com/vladarefiev/timelessis/blob/48bc887089ee05e3cb4f11520901bd151de2134d/timeless/restaurants/models.py#L78

File: tests/integration/sync/it_sync_location_test.py (hunk 5 above)
Comment (vladarefiev) [reply tests/integration/sync/it_sync_location_test.py]: @amihaiemil 
here is - `@unittest.mock.patch.object(Poster, 'locations')`
and here is https://github.com/timelesslounge/timelessis/pull/470/files#diff-e8e21c88402e90db658fb719e93a9369R38

**Comment by vladarefiev** [comment]
@amihaiemil Fixed and answered, take a look pls

File: tests/integration/sync/it_sync_location_test.py (hunk 4 above)
Comment (amihaiemil) [reply tests/integration/sync/it_sync_location_test.py]: @vladarefiev can you then add a message to ``pytest.mark.skip`` to explain the reason? So that we know when to unskip the test (a link to the puzzle/ticket would also be nice). Something like:

``@pytest.mark.skip("Skipped until ticket #xyz (corresponding to that puzzle) is fixed")``

**Comment by amihaiemil** [comment]
@vladarefiev just one more thing about that ``skip`` :D

File: tests/integration/sync/it_sync_location_test.py (hunk 4 above)
Comment (vladarefiev) [reply tests/integration/sync/it_sync_location_test.py]: @amihaiemil Good idea, thanks, will do it now

**Comment by vladarefiev** [comment]
@amihaiemil fixed

**Comment by amihaiemil** [comment]
@vladarefiev nice, thanks :)

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@emilianodellacasa ping

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 36 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#470](https://github.com/timelesslounge/timelessis/issues/470) is now out of scope

**Comment by 0crat** [comment]
Job [#470](https://github.com/timelesslounge/timelessis/issues/470) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm :)

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)
